src/App.tsx

```diff
@@ -1,24 +1,174 @@
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    client.get<Todo[]>(`/todos?userId=${USER_ID}`)
+      .then(setTodos).catch(() => {
+        setError('Unable to load todos');
+        setTimeout(() => {
+          setError('');
+        }, 3000);
+      });
+  }, []);
+
+  const visibleTodos = useMemo(() => {
+    let filteredTodos;
+
+    switch (filterType) {
+      case 'completed':
+        filteredTodos = todos.filter(todo => todo.completed);
+        break;
+      case 'active':
+        filteredTodos = todos.filter(todo => !todo.completed);
+        break;
+      default:
+        filteredTodos = todos;
+        break;
+    }
+
+    return filteredTodos;
+  }, [filterType, todos]);
+
+  const createTodo = async () => {
+    setIsLoading(true);
+    if (!query) {
+      setError('Title can\'t be empty');
+      setTimeout(() => {
+        setError('');
+      }, 3000);
+
+      return;
+    }
+
+    const newTodo: Todo = {
+      id: 0,
+      title: query,
+      userId: USER_ID,
+      completed: false,
+    };
+
+    try {
+      setTemporaryTodo(newTodo);
+
+      const createdTodo = await client.post<Todo>('/todos', newTodo);
+
+      setTodos(prevTodos => [...prevTodos, createdTodo]);
+    } catch {
+      setError('Unable to add a todo');
+      setIsHidden(false);
+      setTimeout(() => {
+        setError('');
+      }, 3000);
+    } finally {
+      setTemporaryTodo(null);
+      setIsHidden(true);
+      setQuery('');
+      setIsLoading(false);
+    }
+  };
+
+  const deleteTodo = async (todoId: number) => {
+    setIsLoading(true);
+    try {
+      await client.delete(`/todos/${todoId}`);
+
+      setTodos(currentTodos => currentTodos.filter(
+        todo => todo.id !== todoId,
+      ));
+    } catch {
+      setError('Unable to delete a todo');
+      setIsHidden(false);
+      setTimeout(() => {
+        setError('');
+      }, 3000);
+    } finally {
+      setIsHidden(true);
+      setIsLoading(false);
+    }
+  };
+
+  const clearCompleted = () => {
+    const chosenTodos = todos.filter(todo => todo.completed === true);
+
+    const clearedTodos = chosenTodos.forEach(
+      todo => client.delete(`/todos/${todo.id}`),
+    );
+
+    setTodos(todos.filter(todo => !chosenTodos.includes(todo)));
+
+    return clearedTodos;
+  };
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">React Todo App - Load Todos</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+
+          {todos.find(todo => !todo.completed) && (
```

Comment:
Let's create a variable for this find. Also, you could use the `useMemo` for storing the value